framework/src/base/Assembly.C

```diff
@@ -402,7 +402,11 @@ Assembly::buildVectorLowerDFE(FEType type) const
   // Build an FE object for this type for each dimension up to the dimension of
   // the current mesh minus one (because this is for lower-dimensional
   // elements!)
-  for (unsigned int dim = 0; dim <= _mesh_dimension - 1; dim++)
+  unsigned int dim = ((type.family == LAGRANGE_VEC) || (type.family == MONOMIAL_VEC)) ? 0 : 2;
```

Comment:
Is the intent here to allow all other vector-valued FE types?  Or should we test for `FEInterface::field_type(type.family) != TYPE_SCALAR` instead of hard-coding?

Reply:
> Is the intent here to allow all other vector-valued FE types?

Yes that is the intent. The choice to allow dim < 2 FEs for LAGRANGE_VEC and MONOMIAL_VEC is somewhat arbitrary and could be removed; I don't think anyone has a use for it. But this most recent code was based of looking at `build` in `FEGenericBase`. Since we allow building those dimensions in libMesh I figured I would allow it here